identity/src/main/java/com/android/identity/util/SimpleApplicationData.java

```diff
@@ -0,0 +1,253 @@
+        }
+        return Util.cborDecodeBoolean(value);
+    }
+
+    /**
+     * Encode the {@link ApplicationData} as a byte[] using <a href="http://cbor.io/">CBOR</a>.
+     *
+     * @return a byte[] of the encoded app data.
+     */
+    public byte[] encodeAsCbor() {
+        CborBuilder appDataBuilder = new CborBuilder();
+        MapBuilder<CborBuilder> appDataMapBuilder = appDataBuilder.addMap();
+        for (String key : mApplicationData.keySet()) {
+            appDataMapBuilder.put(key, mApplicationData.get(key));
+        }
+        appDataMapBuilder.end();
+        return Util.cborEncode(appDataBuilder.build().get(0));
+    }
+
+    /**
+     * Returns a fully populated SimpleApplicationData from a CBOR-encoded byte[].
+     *
+     * <p> To encode a SimpleApplicationData, use {@link #encodeAsCbor()}.
+     *
+     * @param encodedApplicationData The byte array resulting from {@link #encodeAsCbor()}.
+     * @return A SimpleApplicationData with the correct key, value pairs.
+     */
+    public static SimpleApplicationData decodeFromCbor(byte[] encodedApplicationData) {
+        ByteArrayInputStream bais = new ByteArrayInputStream(encodedApplicationData);
+        List<DataItem> dataItems;
+        try {
+            dataItems = new CborDecoder(bais).decode();
+        } catch (CborException e) {
+            throw new IllegalStateException("Error decoded CBOR", e);
+        }
+        if (dataItems.size() != 1) {
+            throw new IllegalStateException("Expected 1 item, found " + dataItems.size());
+        }
+        if (!(dataItems.get(0) instanceof co.nstant.in.cbor.model.Map)) {
+            throw new IllegalStateException("Item is not a map");
+        }
+        DataItem applicationDataDataItem = dataItems.get(0);
+
+        SimpleApplicationData appData = new SimpleApplicationData();
+        for (DataItem keyItem : ((co.nstant.in.cbor.model.Map) applicationDataDataItem).getKeys()) {
+            String key = ((UnicodeString) keyItem).getString();
+            byte[] value = Util.cborMapExtractByteString(applicationDataDataItem, key);
+            appData.setData(key, value);
+        }
+        return appData;
+    }
+
+    /**
+     * Interface for listening to messages from the remote verifier device.
+     *
+     * <p>The Listener#onError(Throwable) callback can be called at any time - for
+     * example - if the remote verifier disconnects without using session termination or if the
+     * underlying transport encounters an unrecoverable error.
+     */
+    public interface ChangeListener {
+
+        /**
+         * Called when an unrecoverable error happens, for example if the remote device
+         * disconnects unexpectedly (e.g. without first sending a session termination request).
+         *
+         * <p>If this is called the application should call disconnect() and the
+         * object should no longer be used.
+         */
+        void onChange();
```

Comment:
Typically past tense is used for listeners so this would be `onChanged()`. Maybe a more accurate name would be `onDataSet()`.